src/webui/api/torrentscontroller.cpp

```diff
@@ -616,9 +616,9 @@ void TorrentsController::addAction()
     const auto autoTMM = TriStateBool::fromString(params()["autoTMM"]);
 
     const QString contentLayoutParam = params()["contentLayout"];
-    const boost::optional<BitTorrent::TorrentContentLayout> contentLayout = (!contentLayoutParam.isEmpty()
+    const std::optional<BitTorrent::TorrentContentLayout> contentLayout = (!contentLayoutParam.isEmpty()
             ? Utils::String::toEnum(contentLayoutParam, BitTorrent::TorrentContentLayout::Original)
-            : boost::optional<BitTorrent::TorrentContentLayout> {});
+            : std::optional<BitTorrent::TorrentContentLayout> {});
```

Comment:
We want both `true`, `false` cases to have the same type, don't we? Then here should be `BitTorrent::TorrentContentLayout {}`.

Reply:
>We want both `true`, `false` cases to have the same type, don't we? 

True branch should return "full" optional.
False branch should return "empty" optional.

Reply:
>We want both `true`, `false` cases to have the same type, don't we? 

The results of both branches of ternary expression are converted to a common type by the compiler (otherwise, how would it be compiled, for example, in this case?). We should only take extra care in cases where there is no unambiguous type casting path, or when there are several common types. Here everything is clear, because `T` and `optional<T>` have a single common type `optional<T>`.